src/qibolab/designs/qmrs.py

```diff
@@ -0,0 +1,160 @@
+from qibo.config import log, raise_error
+
+from qibolab.designs.abstract import AbstractInstrumentDesign
+from qibolab.platforms.utils import Channel
+
+
+class QMRSDesign(AbstractInstrumentDesign):
+    """Instrument design for Quantum Machines (QM) OPXs and Rohde Schwarz local oscillators.
+
+    IPs and other instrument related parameters are hardcoded in ``__init__`` and ``setup``.
+
+    Args:
+        address (str): Address and port for the QM OPX cluster.
+            Default is the current address for the instruments in TII lab.
+        simulation_duration (int): Duration for the simulation in ns.
+            If given the compiler simulator will be used instead of the actual hardware.
+            Default is ``None`` which falls back to the hardware.
+        cloud (bool): See :class:`qibolab.instruments.qmsim.QMSim` for details.
+            Relevant only when ``simulation_duration`` is given.
+
+    Attributes:
+        is_connected (bool): Boolean that shows whether instruments are connected.
+        opx (:class:`qibolab.instruments.qm.QMOPX`): Object used for controlling the QM OPXs.
+        local_oscillators (list): List of local oscillator objects.
+            instrument objects.
+    """
+
+    def __init__(self, address="192.168.0.1:80", simulation_duration=None, cloud=False):
+        from qibolab.instruments.rohde_schwarz import SGS100A
+
+        self.is_connected = False
```

Comment:
This one you can also add to the abstract platform, and invoke `super().__init__()` in every concrete subclass.
I guess it should be there for all platforms, and better not to duplicate shared features (if it is not the case, just tell me and feel free to ignore this comment).